### ISSUE TITLE
[JENKINS-32167] ISVNAuthentication provider did not provide credentials

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/subversion-plugin#1333]
<p>Steps to reproduce the bug:</p>
<ol>
	<li>Install Jenkins 1.617 with Subversion Plug-in 2.5.5</li>
	<li>Declare externals (external_link) on SVN repo</li>
	<li>Create a job that uses a SVN repo with credentials (username/password)</li>
	<li>Setup credential for this repository</li>
	<li>Execute the job</li>
</ol>

<p>Stack trace:</p>

```
hudson.util.IOException2: revision check failed on <external_link>
    at hudson.scm.SubversionChangeLogBuilder.buildModule(SubversionChangeLogBuilder.java:199)
    at hudson.scm.SubversionChangeLogBuilder.run(SubversionChangeLogBuilder.java:137)
    at hudson.scm.SubversionSCM.calcChangeLog(SubversionSCM.java:722)
    at hudson.scm.SubversionSCM.checkout(SubversionSCM.java:857)
    at hudson.scm.SCM.checkout(SCM.java:485)
    at hudson.model.AbstractProject.checkout(AbstractProject.java:1282)
    at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:610)
    at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
    at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:532)
    at hudson.model.Run.execute(Run.java:1744)
    at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
    at hudson.model.ResourceController.execute(ResourceController.java:98)
    at hudson.model.Executor.run(Executor.java:374)

```

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mariobros">mariobros</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167" target="_blank">ISVNAuthentication provider did not provide credentials</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick">jglick</a>
<li><b>status</b>: Fixed but Unreleased
<li><b>priority</b>: Major
<li><b>component(s)</b>: subversion-plugin
<li><b>label(s)</b>: credentials, diagnostics
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2018-09-12T21:48:15+00:00
<li><b>votes</b>: 65
<li><b>watchers</b>: 78
<li><b>imported</b>: 20251212-090250
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Steps to reproduce the bug:</p>
<ol>
	<li>Install Jenkins 1.617 with Subversion Plug-in 2.5.5</li>
	<li>Declare externals (external_link) on SVN repo</li>
	<li>Create a job that uses a SVN repo with credentials (username/password)</li>
	<li>Setup credential for this repository</li>
	<li>Execute the job</li>
</ol>

<p>Stack trace:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>hudson.util.IOException2: revision check failed on &lt;external_link&gt;
    at hudson.scm.SubversionChangeLogBuilder.buildModule(SubversionChangeLogBuilder.java:199)
    at hudson.scm.SubversionChangeLogBuilder.run(SubversionChangeLogBuilder.java:137)
    at hudson.scm.SubversionSCM.calcChangeLog(SubversionSCM.java:722)
    at hudson.scm.SubversionSCM.checkout(SubversionSCM.java:857)
    at hudson.scm.SCM.checkout(SCM.java:485)
    at hudson.model.AbstractProject.checkout(AbstractProject.java:1282)
    at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:610)
    at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
    at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:532)
    at hudson.model.Run.execute(Run.java:1744)
    at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
    at hudson.model.ResourceController.execute(ResourceController.java:98)
    at hudson.model.Executor.run(Executor.java:374)
</pre>
</div></div></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Windows 7; Jenkins 1.617 with Subversion Plug-in 2.5.5<br/>
Windows Server 2012 R2; Jenkins 2.19.3 with Subversion plugin 2.7.1<br/>
Linux x86_64, Jenkins 2.19.4 with Subversion Plugin 2.7.1<br/>
```
</details>
<details><summary><i>6 attachments</i></summary>

- [CredentialsExternalsTest.java](https://issues.jenkins.io/secure/attachment/37601/CredentialsExternalsTest.java)
- [Jenkins config.png](https://issues.jenkins.io/secure/attachment/41343/Jenkins%20config.png)
  > ![Jenkins config.png](https://issues.jenkins.io/secure/attachment/41343/Jenkins%20config.png)
- [Job config.png](https://issues.jenkins.io/secure/attachment/41342/Job%20config.png)
  > ![Job config.png](https://issues.jenkins.io/secure/attachment/41342/Job%20config.png)
- [Screenshot_Jenkins_Config.png](https://issues.jenkins.io/secure/attachment/41348/Screenshot_Jenkins_Config.png)
  > ![Screenshot_Jenkins_Config.png](https://issues.jenkins.io/secure/attachment/41348/Screenshot_Jenkins_Config.png)
- [screenshot01.png](https://issues.jenkins.io/secure/attachment/35832/screenshot01.png)
  > ![screenshot01.png](https://issues.jenkins.io/secure/attachment/35832/screenshot01.png)
- [screenshot02.png](https://issues.jenkins.io/secure/attachment/35831/screenshot02.png)
  > ![screenshot02.png](https://issues.jenkins.io/secure/attachment/35831/screenshot02.png)
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote links associated with this issue:_


  - [PR 177](https://github.com/jenkinsci/subversion-plugin/pull/177)
  - [CloudBees Internal OSS-1490](https://cloudbees.atlassian.net/browse/OSS-1490)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jakste:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=246554&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-246554">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Works here to... Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=advorsky73" class="user-hover" rel="advorsky73">advorsky73</a> !</p>
</pre>
</details>

</ul>
</details>

<p>Works here to... Thanks <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=advorsky73" class="user-hover" rel="advorsky73">advorsky73</a> !</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>vanista:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=247110&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-247110">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><del>The workaround is not quite enough for us; when switching repos for a tag build, the externals are not checked out/updated properly.</del></p>

<p>Nevermind that, upon closer inspection it's a problem with svn itself when switching repos and with an external that is replaced by a directory, so unrelated to original topic.</p>

<p>The patched plugin still works fine for us.</p>
</pre>
</details>

</ul>
</details>

<p><del>The workaround is not quite enough for us; when switching repos for a tag build, the externals are not checked out/updated properly.</del></p>

<p>Nevermind that, upon closer inspection it's a problem with svn itself when switching repos and with an external that is replaced by a directory, so unrelated to original topic.</p>

<p>The patched plugin still works fine for us.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/advorsky73.jpg" title="advorsky73's avatar" /> advorsky73:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=247381&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-247381">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Manuel Jesús Recena Soto are you already able to reproduce the issue?
If not, please let me know and i will set up a small sandbox svn repo and dump it for you to use...</p>

<p>Maciej Jaros &gt; I see no changes in build details and SVN version doesn't include the external, but I don't think it ever did anyway.
Actually the external changes were included in the svn changelog in jenkins if i remember correctly.</p>
</pre>
</details>

</ul>
</details>

<p>Manuel Jesús Recena Soto are you already able to reproduce the issue?<br/>
If not, please let me know and i will set up a small sandbox svn repo and dump it for you to use...</p>

<p>Maciej Jaros > I see no changes in build details and SVN version doesn't include the external, but I don't think it ever did anyway.<br/>
Actually the external changes were included in the svn changelog in jenkins if i remember correctly.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/advorsky73.jpg" title="advorsky73's avatar" /> advorsky73:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=247672&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-247672">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>New Download Link, file vanished: <a href="http://filestore.to/?d=NODV33FVLR" class="external-link" target="_blank" rel="nofollow noopener">http://filestore.to/?d=NODV33FVLR</a> </p>
</pre>
</details>

</ul>
</details>

<p>New Download Link, file vanished: <a href="http://filestore.to/?d=NODV33FVLR" class="external-link" target="_blank" rel="nofollow noopener">http://filestore.to/?d=NODV33FVLR</a> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/advorsky73.jpg" title="advorsky73's avatar" /> advorsky73:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=248172&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-248172">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm a bit curious... did you get the file or shall i upload it again Manuel Jesús Recena Soto ?</p>
</pre>
</details>

</ul>
</details>

<p>I'm a bit curious... did you get the file or shall i upload it again Manuel Jesús Recena Soto ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>sanrudnev:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=248173&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-248173">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is there a workaround for this issue? I changed my Jenkins version from 156x to the latest and the bug is <b>very</b> annoying. Unfortunately I can't roll back to older version, as they are not backward compatible. That is P1 for me, really.</p>
</pre>
</details>

</ul>
</details>

<p>Is there a workaround for this issue? I changed my Jenkins version from 156x to the latest and the bug is <b>very</b> annoying. Unfortunately I can't roll back to older version, as they are not backward compatible. That is P1 for me, really.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nutmix:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=248174&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-248174">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=sanrudnev" class="user-hover" rel="sanrudnev">sanrudnev</a> We have not found a work around yet - other than to always run the build twice.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=sanrudnev" class="user-hover" rel="sanrudnev">sanrudnev</a> We have not found a work around yet - other than to always run the build twice.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/advorsky73.jpg" title="advorsky73's avatar" /> advorsky73:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=249133&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-249133">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Well time passes without anything happening here...
Any progress on this, or can i be of any assistance?</p>

<p>btw, where can i find a how to for debugging a plugin for jenkins? any hints/links appreciated.</p>
</pre>
</details>

</ul>
</details>

<p>Well time passes without anything happening here...<br/>
Any progress on this, or can i be of any assistance?</p>

<p>btw, where can i find a how to for debugging a plugin for jenkins? any hints/links appreciated.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>sanrudnev:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=249163&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-249163">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Why the issue is closed? The fact that workaround exists doesn't make the bug solved, it just lowers the priority.</p>
</pre>
</details>

</ul>
</details>

<p>Why the issue is closed? The fact that workaround exists doesn't make the bug solved, it just lowers the priority.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>fnxweb:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=249275&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-249275">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>No, I'm sorry, this is making no sense to me;  my Jenkins (1.625.3 - LTS) is showing me similar options to Alexander's not Jörg's, so I'm not sure where the realm reference is supposed to come from.  This is what I have in a typical job (note:  I'm only using one SVN server, for main and externals, so I've only been giving one credential entry, and that comes from the core Jenkins credentials setup):
<em>(job config)</em></p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Subversion
	Modules	
		Repository URL		https://myservername/myrepourl
		Credentials		username/****** (Guest login)  [from drop down menu]
		Local module directory	
		Repository depth	infinity
</pre>
</div></div>

<p><em>(main Jenkins config)</em></p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Credentials
	Scope		Global (Jenkins, nodes, items, all child items, etc)
 	Username	username
 	Password	***************************
 	Description	Guest login
		[Help: An optional description to help tell similar credentials apart.]
</pre>
</div></div>

<p>Where does realm come into this?
Do I need to now add those credentials as additional credentials to the job, even though they're the same (and I haven't had to before)?
Why does my job-credentials drop-down menu show my what Alexander's does (“username/password description-text”) and not what others are seeing (realm info.)?</p>

<p>Note also I'm currently using Alexander's patched version of the plugin (thanks Alexander!).</p>
</pre>
</details>

</ul>
</details>

<p>No, I'm sorry, this is making no sense to me;  my Jenkins (1.625.3 - LTS) is showing me similar options to Alexander's not Jörg's, so I'm not sure where the realm reference is supposed to come from.  This is what I have in a typical job (note:  I'm only using one SVN server, for main and externals, so I've only been giving one credential entry, and that comes from the core Jenkins credentials setup):<br/>
<em>(job config)</em></p>


```
Subversion
	Modules	
		Repository URL		https://myservername/myrepourl
		Credentials		username/****** (Guest login)  [from drop down menu]
		Local module directory	
		Repository depth	infinity

```

<p><em>(main Jenkins config)</em></p>


```
Credentials
	Scope		Global (Jenkins, nodes, items, all child items, etc)
 	Username	username
 	Password	***************************
 	Description	Guest login
		[Help: An optional description to help tell similar credentials apart.]

```

<p>Where does realm come into this?<br/>
Do I need to now add those credentials as additional credentials to the job, even though they're the same (and I haven't had to before)?<br/>
Why does my job-credentials drop-down menu show my what Alexander's does (“username/password description-text”) and not what others are seeing (realm info.)?</p>

<p>Note also I'm currently using Alexander's patched version of the plugin (thanks Alexander!).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/advorsky73.jpg" title="advorsky73's avatar" /> advorsky73:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=249306&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-249306">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>OK step by step guide to avoid this stupid SVN Plugin Error....
And YES THAT IS STILL AN ERROR... uncaught exceptions without meaningful messages are ERRORs....even when caused by wrong input of the users....
So please reopen that ISSUE and make a PROPER FIX!</p>

<p>1. Get your realm string (NOT NAME, you need the WHOLE STRING!)</p>

<p>Execute the following cmd line with &lt;URL&gt; replaced by your SVN url. Do NOT use an IP address, use a DNS name !!!</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>svn --no-auth-cache --config-dir invalid info http://&lt;URL&gt;
Authentication realm: &lt;http://&lt;URL&gt;:port&gt; SomeTextHere
</pre>
</div></div>

<p>The output should be the following:</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>Authentication realm: &lt;http://&lt;URL&gt;:port&gt; SomeTextHere
</pre>
</div></div>

<p>Your REALM STRING is then "&lt;http://&lt;URL&gt;:port&gt; SomeTextHere", so basically EVERYTHING after "Authentication realm: " !!!</p>

<p>2. Create a credential in Jenkins with your REALM STRING in the DESCRIPTION FIELD !!!</p>

<p>3. In your job add an additional credential with the button "Add additional credentials..."</p>

<p>4. Put your REALM STRING in there and select the previous created credential with the SAME REALM STRING in the description</p>

<p>5. Enjoy a crash free execution with history of the external in your builds.</p>

<p>Now somebody PLEASE put that in the documentation or in a help section.
Thanks.</p>
</pre>
</details>

</ul>
</details>

<p>OK step by step guide to avoid this stupid SVN Plugin Error....<br/>
And YES THAT IS STILL AN ERROR... uncaught exceptions without meaningful messages are ERRORs....even when caused by wrong input of the users....<br/>
So please reopen that ISSUE and make a PROPER FIX!</p>

<p>1. Get your realm string (NOT NAME, you need the WHOLE STRING!)</p>

<p>Execute the following cmd line with <URL> replaced by your SVN url. Do NOT use an IP address, use a DNS name !!!</p>


```
svn --no-auth-cache --config-dir invalid info http://<URL>
Authentication realm: <http://<URL>:port> SomeTextHere

```

<p>The output should be the following:</p>


```
Authentication realm: <http://<URL>:port> SomeTextHere

```

<p>Your REALM STRING is then "<http://<URL>:port> SomeTextHere", so basically EVERYTHING after "Authentication realm: " !!!</p>

<p>2. Create a credential in Jenkins with your REALM STRING in the DESCRIPTION FIELD !!!</p>

<p>3. In your job add an additional credential with the button "Add additional credentials..."</p>

<p>4. Put your REALM STRING in there and select the previous created credential with the SAME REALM STRING in the description</p>

<p>5. Enjoy a crash free execution with history of the external in your builds.</p>

<p>Now somebody PLEASE put that in the documentation or in a help section.<br/>
Thanks.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>nutmix:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=249308&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-249308">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks for the update - this kind of detailed documentation is desperately missing.</p>

<p>Sadly, it still does not work. before updating SVN plugin, worked fine, after updating SVN plugin, gives this error.</p>

<p>Note, when I add "Additional credentials", there is only "Realm" and "Credentials" boxes, there is no box for "DESCRIPTION FIELD".</p>

<p>My realm happens to be:</p>

<p>&lt;<a href="https://subversion.assembla.com:443" class="external-link" target="_blank" rel="nofollow noopener">https://subversion.assembla.com:443</a>&gt; Assembla Restricted</p>

</pre>
</details>

</ul>
</details>

<p>Thanks for the update - this kind of detailed documentation is desperately missing.</p>

<p>Sadly, it still does not work. before updating SVN plugin, worked fine, after updating SVN plugin, gives this error.</p>

<p>Note, when I add "Additional credentials", there is only "Realm" and "Credentials" boxes, there is no box for "DESCRIPTION FIELD".</p>

<p>My realm happens to be:</p>

<p><<a href="https://subversion.assembla.com:443" class="external-link" target="_blank" rel="nofollow noopener">https://subversion.assembla.com:443</a>> Assembla Restricted</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/advorsky73.jpg" title="advorsky73's avatar" /> advorsky73:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=249309&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-249309">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi John...</p>

<p>it is all kind of missleading because its almost named the same...
Click the ADD button right beside the "Credentials" Combo/Dropdown Box. 
That's where you ADD the credential as mentioned in step 2 first, so the name in the combo/dropdown box looks like expected.</p>

<p>cheers,
Alex</p>
</pre>
</details>

</ul>
</details>

<p>Hi John...</p>

<p>it is all kind of missleading because its almost named the same...<br/>
Click the ADD button right beside the "Credentials" Combo/Dropdown Box. <br/>
That's where you ADD the credential as mentioned in step 2 first, so the name in the combo/dropdown box looks like expected.</p>

<p>cheers,<br/>
Alex</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/recena.png" title="recena's avatar" /> recena:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=249343&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-249343">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Everything here has been explained in similar issues.</p>
</pre>
</details>

</ul>
</details>

<p>Everything here has been explained in similar issues.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/mikelo.png" title="mikelo's avatar" /> mikelo:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=249348&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-249348">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Can you please link the similar issues? Maybe it gives us some insight. If someone files new issues related to this, please let us know in this thread.</p>

<p>I had no luck configuring the different realms, so I have a question. I have configured many jobs for different repositories and each on of them have several externals. All the repositories and externals share the same password database. So, do I have to configure all the credentials for all the different realms? This sounds somehow cumbersome... Actually, the plugin works if you do not specify the realm, but it gives an exception. I think that exception should be analyzed or treated in a different way. I mean, running the job twice works without specifying the realm, shouldn't it always fail if you do not specify the realm?</p>
</pre>
</details>

</ul>
</details>

<p>Can you please link the similar issues? Maybe it gives us some insight. If someone files new issues related to this, please let us know in this thread.</p>

<p>I had no luck configuring the different realms, so I have a question. I have configured many jobs for different repositories and each on of them have several externals. All the repositories and externals share the same password database. So, do I have to configure all the credentials for all the different realms? This sounds somehow cumbersome... Actually, the plugin works if you do not specify the realm, but it gives an exception. I think that exception should be analyzed or treated in a different way. I mean, running the job twice works without specifying the realm, shouldn't it always fail if you do not specify the realm?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>tini:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=249434&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-249434">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>SQE here.  I signed up for this account just to EMPHATICALLY AGREE with @advorsky73.</p>

<p>I've read dozens of these issues with this exact same problem trying to figure out why we are constantly getting these stupid errors when we have everything done correctly only to discover that it's not enough to provide the Realm, it wants the stupid text after it??</p>

<p>This is defective error handling AND documentation that wastes your time and ours.  Please fix. </p>
</pre>
</details>

</ul>
</details>

<p>SQE here.  I signed up for this account just to EMPHATICALLY AGREE with @​advorsky73.</p>

<p>I've read dozens of these issues with this exact same problem trying to figure out why we are constantly getting these stupid errors when we have everything done correctly only to discover that it's not enough to provide the Realm, it wants the stupid text after it??</p>

<p>This is defective error handling AND documentation that wastes your time and ours.  Please fix. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/chandau.png" title="chandau's avatar" /> chandau:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=309832&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-309832">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>just want to add that this is happening on Jenkins v 2.60.2 and 2.72, subversion plugin version 2.9</p>
</pre>
</details>

</ul>
</details>

<p>just want to add that this is happening on Jenkins v 2.60.2 and 2.72, subversion plugin version 2.9</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/walteste.jpg" title="walteste's avatar" /> walteste:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=315610&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315610">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is there any progress or any idea how this will be solved? Is there an alternative to this plugin?</p>
</pre>
</details>

</ul>
</details>

<p>Is there any progress or any idea how this will be solved? Is there an alternative to this plugin?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jglick.png" title="jglick's avatar" /> jglick:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=316536&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-316536">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p>Is there any progress</p></blockquote>
<p>Afraid not.</p>
<blockquote><p>Is there an alternative to this plugin?</p></blockquote>
<p>Running <tt>svn up</tt> from a shell step in a job, and giving up on polling, changelogs, multibranch, etc.</p>
</pre>
</details>

</ul>
</details>

<blockquote><p>Is there any progress</p></blockquote>
<p>Afraid not.</p>
<blockquote><p>Is there an alternative to this plugin?</p></blockquote>
<p>Running <tt>svn up</tt> from a shell step in a job, and giving up on polling, changelogs, multibranch, etc.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>larryhou:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=324038&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-324038">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I encountered this problem too</p>

<p>Jenkins v2.98</p>

<p>Subversion Plugin v2.10.2</p>


<p> org.tmatesoft.svn.core.SVNCancelException: svn: E200015: ISVNAuthentication provider did not provide credentials; HTTP authorization cancelled.
 at org.tmatesoft.svn.core.internal.io.dav.http.HTTPConnection.request(HTTPConnection.java:728)
 Caused: org.tmatesoft.svn.core.SVNCancelException: svn: E200015: E200015: ISVNAuthentication provider did not provide credentials; HTTP authorization cancelled.
 svn: E200015: ISVNAuthentication provider did not provide credentials; HTTP authorization cancelled.
 at org.tmatesoft.svn.core.internal.wc.SVNErrorManager.error(SVNErrorManager.java:66)
 at org.tmatesoft.svn.core.internal.wc.SVNErrorManager.error(SVNErrorManager.java:57)
 at org.tmatesoft.svn.core.internal.io.dav.http.HTTPConnection.request(HTTPConnection.java:798)
 at org.tmatesoft.svn.core.internal.io.dav.http.HTTPConnection.request(HTTPConnection.java:391)
 at org.tmatesoft.svn.core.internal.io.dav.http.HTTPConnection.request(HTTPConnection.java:379)
 at org.tmatesoft.svn.core.internal.io.dav.DAVConnection.performHttpRequest(DAVConnection.java:862)
 at org.tmatesoft.svn.core.internal.io.dav.DAVConnection.exchangeCapabilities(DAVConnection.java:698)
 at org.tmatesoft.svn.core.internal.io.dav.DAVConnection.open(DAVConnection.java:118)
 at org.tmatesoft.svn.core.internal.io.dav.DAVRepository.openConnection(DAVRepository.java:1049)
 at org.tmatesoft.svn.core.internal.io.dav.DAVRepository.getLatestRevision(DAVRepository.java:189)
 at org.tmatesoft.svn.core.internal.wc2.ng.SvnNgRepositoryAccess.getRevisionNumber(SvnNgRepositoryAccess.java:119)
 at org.tmatesoft.svn.core.internal.wc2.SvnRepositoryAccess.getLocations(SvnRepositoryAccess.java:195)
 at org.tmatesoft.svn.core.internal.wc2.ng.SvnNgRepositoryAccess.createRepositoryFor(SvnNgRepositoryAccess.java:46)
 at org.tmatesoft.svn.core.internal.wc2.remote.SvnRemoteLog.run(SvnRemoteLog.java:160)
 at org.tmatesoft.svn.core.internal.wc2.remote.SvnRemoteLog.run(SvnRemoteLog.java:35)
 at org.tmatesoft.svn.core.internal.wc2.SvnOperationRunner.run(SvnOperationRunner.java:21)
 at org.tmatesoft.svn.core.wc2.SvnOperationFactory.run(SvnOperationFactory.java:1235)
 at org.tmatesoft.svn.core.wc2.SvnOperation.run(SvnOperation.java:294)
 at org.tmatesoft.svn.core.wc.SVNLogClient.doLog(SVNLogClient.java:968)
 at org.tmatesoft.svn.core.wc.SVNLogClient.doLog(SVNLogClient.java:873)
 at hudson.scm.SubversionChangeLogBuilder.buildModule(SubversionChangeLogBuilder.java:194)
 Caused: hudson.util.IOException2: revision check failed on <a href="http://tc-svn.tencent.com/ied/ied_narutoNext_rep/naruto_next_proj/trunk/TheNextMOBA/Assets/Plugins/MorefunLockStep.dll" class="external-link" target="_blank" rel="nofollow noopener">http://tc-svn.tencent.com/ied/ied_narutoNext_rep/naruto_next_proj/trunk/TheNextMOBA/Assets/Plugins/MorefunLockStep.dll</a>
 at hudson.scm.SubversionChangeLogBuilder.buildModule(SubversionChangeLogBuilder.java:208)
 at hudson.scm.SubversionChangeLogBuilder.run(SubversionChangeLogBuilder.java:138)
 at hudson.scm.SubversionSCM.calcChangeLog(SubversionSCM.java:756)
 at hudson.scm.SubversionSCM.checkout(SubversionSCM.java:891)
 at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:113)
 at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:85)
 at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:75)
 at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)
 at hudson.security.ACL.impersonate(ACL.java:274)
 at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)
 at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
 at java.util.concurrent.FutureTask.run(FutureTask.java:266)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
 at java.lang.Thread.run(Thread.java:745)</p>
</pre>
</details>

</ul>
</details>

<p>I encountered this problem too</p>

<p>Jenkins v2.98</p>

<p>Subversion Plugin v2.10.2</p>


<p> org.tmatesoft.svn.core.SVNCancelException: svn: E200015: ISVNAuthentication provider did not provide credentials; HTTP authorization cancelled.<br/>
 at org.tmatesoft.svn.core.internal.io.dav.http.HTTPConnection.request(HTTPConnection.java:728)<br/>
 Caused: org.tmatesoft.svn.core.SVNCancelException: svn: E200015: E200015: ISVNAuthentication provider did not provide credentials; HTTP authorization cancelled.<br/>
 svn: E200015: ISVNAuthentication provider did not provide credentials; HTTP authorization cancelled.<br/>
 at org.tmatesoft.svn.core.internal.wc.SVNErrorManager.error(SVNErrorManager.java:66)<br/>
 at org.tmatesoft.svn.core.internal.wc.SVNErrorManager.error(SVNErrorManager.java:57)<br/>
 at org.tmatesoft.svn.core.internal.io.dav.http.HTTPConnection.request(HTTPConnection.java:798)<br/>
 at org.tmatesoft.svn.core.internal.io.dav.http.HTTPConnection.request(HTTPConnection.java:391)<br/>
 at org.tmatesoft.svn.core.internal.io.dav.http.HTTPConnection.request(HTTPConnection.java:379)<br/>
 at org.tmatesoft.svn.core.internal.io.dav.DAVConnection.performHttpRequest(DAVConnection.java:862)<br/>
 at org.tmatesoft.svn.core.internal.io.dav.DAVConnection.exchangeCapabilities(DAVConnection.java:698)<br/>
 at org.tmatesoft.svn.core.internal.io.dav.DAVConnection.open(DAVConnection.java:118)<br/>
 at org.tmatesoft.svn.core.internal.io.dav.DAVRepository.openConnection(DAVRepository.java:1049)<br/>
 at org.tmatesoft.svn.core.internal.io.dav.DAVRepository.getLatestRevision(DAVRepository.java:189)<br/>
 at org.tmatesoft.svn.core.internal.wc2.ng.SvnNgRepositoryAccess.getRevisionNumber(SvnNgRepositoryAccess.java:119)<br/>
 at org.tmatesoft.svn.core.internal.wc2.SvnRepositoryAccess.getLocations(SvnRepositoryAccess.java:195)<br/>
 at org.tmatesoft.svn.core.internal.wc2.ng.SvnNgRepositoryAccess.createRepositoryFor(SvnNgRepositoryAccess.java:46)<br/>
 at org.tmatesoft.svn.core.internal.wc2.remote.SvnRemoteLog.run(SvnRemoteLog.java:160)<br/>
 at org.tmatesoft.svn.core.internal.wc2.remote.SvnRemoteLog.run(SvnRemoteLog.java:35)<br/>
 at org.tmatesoft.svn.core.internal.wc2.SvnOperationRunner.run(SvnOperationRunner.java:21)<br/>
 at org.tmatesoft.svn.core.wc2.SvnOperationFactory.run(SvnOperationFactory.java:1235)<br/>
 at org.tmatesoft.svn.core.wc2.SvnOperation.run(SvnOperation.java:294)<br/>
 at org.tmatesoft.svn.core.wc.SVNLogClient.doLog(SVNLogClient.java:968)<br/>
 at org.tmatesoft.svn.core.wc.SVNLogClient.doLog(SVNLogClient.java:873)<br/>
 at hudson.scm.SubversionChangeLogBuilder.buildModule(SubversionChangeLogBuilder.java:194)<br/>
 Caused: hudson.util.IOException2: revision check failed on <a href="http://tc-svn.tencent.com/ied/ied_narutoNext_rep/naruto_next_proj/trunk/TheNextMOBA/Assets/Plugins/MorefunLockStep.dll" class="external-link" target="_blank" rel="nofollow noopener">http://tc-svn.tencent.com/ied/ied_narutoNext_rep/naruto_next_proj/trunk/TheNextMOBA/Assets/Plugins/MorefunLockStep.dll</a><br/>
 at hudson.scm.SubversionChangeLogBuilder.buildModule(SubversionChangeLogBuilder.java:208)<br/>
 at hudson.scm.SubversionChangeLogBuilder.run(SubversionChangeLogBuilder.java:138)<br/>
 at hudson.scm.SubversionSCM.calcChangeLog(SubversionSCM.java:756)<br/>
 at hudson.scm.SubversionSCM.checkout(SubversionSCM.java:891)<br/>
 at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:113)<br/>
 at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:85)<br/>
 at org.jenkinsci.plugins.workflow.steps.scm.SCMStep$StepExecutionImpl.run(SCMStep.java:75)<br/>
 at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1$1.call(AbstractSynchronousNonBlockingStepExecution.java:47)<br/>
 at hudson.security.ACL.impersonate(ACL.java:274)<br/>
 at org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution$1.run(AbstractSynchronousNonBlockingStepExecution.java:44)<br/>
 at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)<br/>
 at java.util.concurrent.FutureTask.run(FutureTask.java:266)<br/>
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)<br/>
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)<br/>
 at java.lang.Thread.run(Thread.java:745)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>smouch:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=325806&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-325806">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>If you search for my comments on other related issues, you will discover that really, the only solution to this issue is one which the Jenkins maintainers refuse to even consider.</p>

<p> </p>

<p>The built-in SVN "plug-in" (not really a plug-in since it's built in) is just a bad idea.  All SVN commands should simply be forwarded to the slave running the job.  Of course this means that the slave needs to have SVN installed, but then, it needs to have other tools for the build installed too, so, where's the problem ?</p>
</pre>
</details>

</ul>
</details>

<p>If you search for my comments on other related issues, you will discover that really, the only solution to this issue is one which the Jenkins maintainers refuse to even consider.</p>

<p> </p>

<p>The built-in SVN "plug-in" (not really a plug-in since it's built in) is just a bad idea.  All SVN commands should simply be forwarded to the slave running the job.  Of course this means that the slave needs to have SVN installed, but then, it needs to have other tools for the build installed too, so, where's the problem ?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>gl1koz3:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=327129&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-327129">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Using built-in tools can help in some scenarios, but doesn't solve it all. Jenkins still needs a way to communicate with version control without re-spawning thousands of processes. Plus, managing tools and their command lines would be another pain, including the fact that there are many tools for every version control. It would just shift the problem to another place.</p>

<p>Plus, you can already invoke your tools from shell, so what's even the point?</p>

<p>The plugin should get fixed.</p>
</pre>
</details>

</ul>
</details>

<p>Using built-in tools can help in some scenarios, but doesn't solve it all. Jenkins still needs a way to communicate with version control without re-spawning thousands of processes. Plus, managing tools and their command lines would be another pain, including the fact that there are many tools for every version control. It would just shift the problem to another place.</p>

<p>Plus, you can already invoke your tools from shell, so what's even the point?</p>

<p>The plugin should get fixed.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>barilskag:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=327694&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-327694">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi, I have the following problem- the jobs are configured with Jenkins credentials and it's working well, but in Jenkins configuration for SCM sync configuration there is not an option to add the credentials, so I'm receiving the same error. Did you have any success with this issue?</p>

<p>This is how it looks the Job configuration <font color="#ff0000">Job config.png</font>  (-&gt; It provides a place for setting credentials)</p>

<p>And here is the Manage Jenkins  <font color="#ff0000">Jenkins config.png</font> -&gt; Configure System: ( It's not provided a place to add credentials for SCM sync Plugin here so it leads to
 "SEVERE: svn: E200015: E200015: ISVNAuthentication provider did not provide credentials; HTTP authorization cancelled.")</p>

<p>(The screenshots are attached)</p>

<p>Can anyone help with configuring SCM authenticatuion?</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>Hi, I have the following problem- the jobs are configured with Jenkins credentials and it's working well, but in Jenkins configuration for SCM sync configuration there is not an option to add the credentials, so I'm receiving the same error. Did you have any success with this issue?</p>

<p>This is how it looks the Job configuration <font color="#ff0000">Job config.png</font>  (-> It provides a place for setting credentials)</p>

<p>And here is the Manage Jenkins  <font color="#ff0000">Jenkins config.png</font> -> Configure System: ( It's not provided a place to add credentials for SCM sync Plugin here so it leads to<br/>
 "SEVERE: svn: E200015: E200015: ISVNAuthentication provider did not provide credentials; HTTP authorization cancelled.")</p>

<p>(The screenshots are attached)</p>

<p>Can anyone help with configuring SCM authenticatuion?</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>smccants:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=327749&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-327749">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=barilskag" class="user-hover" rel="barilskag">barilskag</a> looks like there is something missing in your UI that is making this problem more difficult.  I'm running an old version of Jenkins (2.49) and in my Job Configuration screen have a "Add Additional Credentials" button.  If that button is missing, I suspect that is a separate issue from this one (but one preventing you from being able to use the work arounds described above).  I've attached a screen shot as reference.  Hope that helps. <span class="image-wrap" style=""><img src="https://issues.jenkins.io/secure/attachment/41348/41348_Screenshot_Jenkins_Config.png" style="border: 0px solid black" /></span></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=barilskag" class="user-hover" rel="barilskag">barilskag</a> looks like there is something missing in your UI that is making this problem more difficult.  I'm running an old version of Jenkins (2.49) and in my Job Configuration screen have a "Add Additional Credentials" button.  If that button is missing, I suspect that is a separate issue from this one (but one preventing you from being able to use the work arounds described above).  I've attached a screen shot as reference.  Hope that helps. <span class="image-wrap" style=""><img src="https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-last/refs/heads/main/attachments/41348/Screenshot_Jenkins_Config.png" style="border: 0px solid black" /></span></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>smccants:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=333912&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-333912">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gl1koz3" class="user-hover" rel="gl1koz3">gl1koz3</a>, unfortunately no one owns or maintains this code, so there is no ETA.  Wish I had better news for all of us who are affected.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gl1koz3" class="user-hover" rel="gl1koz3">gl1koz3</a>, unfortunately no one owns or maintains this code, so there is no ETA.  Wish I had better news for all of us who are affected.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/nux.png" title="nux's avatar" /> nux:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=334225&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-334225">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I used <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=advorsky73" class="user-hover" rel="advorsky73">advorsky73</a> solution from <a href="#comment-246389" target="_blank" rel="noopener">comment-246389</a> since January 2016. Didn't update Jenkins (still on 1.x), but 2.5.7 version of SVN plugin with that fix works fine. All features seem to work correctly (checkout works with and without externals, changes show up in the build, passing credentials work). Most of the builds in installation of my current company use SVN. And there are many externals in repos used.</p>

<p>It would seem the workaround is a valid solution (as there has been no other solution present) and the exception is a false negative somehow... Could someone push the solution to the official repo and run some automated tests (if there are any for this one).</p>
</pre>
</details>

</ul>
</details>

<p>I used <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=advorsky73" class="user-hover" rel="advorsky73">advorsky73</a> solution from <a href="#comment-246389" target="_blank" rel="noopener">comment-246389</a> since January 2016. Didn't update Jenkins (still on 1.x), but 2.5.7 version of SVN plugin with that fix works fine. All features seem to work correctly (checkout works with and without externals, changes show up in the build, passing credentials work). Most of the builds in installation of my current company use SVN. And there are many externals in repos used.</p>

<p>It would seem the workaround is a valid solution (as there has been no other solution present) and the exception is a false negative somehow... Could someone push the solution to the official repo and run some automated tests (if there are any for this one).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mgrochowalski:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=334531&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-334531">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Liyaa submitted a pull request to fix this issue:</p>

<p><a href="https://github.com/jenkinsci/subversion-plugin/pull/212" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/subversion-plugin/pull/212</a></p>
</pre>
</details>

</ul>
</details>

<p>Liyaa submitted a pull request to fix this issue:</p>

<p><a href="https://github.com/jenkinsci/subversion-plugin/pull/212" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/subversion-plugin/pull/212</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>alucas01:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=335566&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-335566">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>You closed <a href="https://issues.jenkins.io/browse/JENKINS-31455" title="Build instability with &quot;ISVNAuthentication provider did not provide credentials&quot;" class="issue-link" data-issue-key="JENKINS-31455"><del>JENKINS-31455</del></a> as a duplicate to this one. If this one ever gets worked, be sure to address that it is seen when using external svn references in a build. The first build fails with the authentication error, then the next completes. The cause of the build is lost to the second build, and two builds have to occur.</p>

<p>31455 had some comments that might be useful to keep track of:</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=smccants" class="user-hover" rel="smccants">smccants</a> added a comment - 2018-02-13 13:59
<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dimitkoto" class="user-hover" rel="dimitkoto">dimitkoto</a>, as stated in <font color="#000080"><a href="https://issues.jenkins.io/browse/JENKINS-32167" title="ISVNAuthentication provider did not provide credentials" class="issue-link" data-issue-key="JENKINS-32167"><del>JENKINS-32167</del></a></font>, there is no owner or maintainer of this plugin.  There is a workaround provided in comments under <font color="#000080"><a href="https://issues.jenkins.io/browse/JENKINS-32167" title="ISVNAuthentication provider did not provide credentials" class="issue-link" data-issue-key="JENKINS-32167"><del>JENKINS-32167</del></a></font>.  <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> provided some better feedback code, but I don't think that has ever shipped.  I looked at it and wrote a test case demonstrating the problem, but solving it was too complicated for the time I could devote to it.</p>

<p>I don't think anything has changed since then.   Wish someone would step up and fix it for us users, but I'm not expecting it until the SVN plugin is totally busted.</p>

<p> </p>
</pre>
</details>

</ul>
</details>

<p>You closed <a href="https://issue-redirect.jenkins.io/issue/31455" title="Build instability with "ISVNAuthentication provider did not provide credentials"" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/31455">JENKINS-31455</a>"><del>JENKINS-31455</del></a> as a duplicate to this one. If this one ever gets worked, be sure to address that it is seen when using external svn references in a build. The first build fails with the authentication error, then the next completes. The cause of the build is lost to the second build, and two builds have to occur.</p>

<p>31455 had some comments that might be useful to keep track of:</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=smccants" class="user-hover" rel="smccants">smccants</a> added a comment - 2018-02-13 13:59<br/>
<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=dimitkoto" class="user-hover" rel="dimitkoto">dimitkoto</a>, as stated in <font color="#000080"><a href="https://issue-redirect.jenkins.io/issue/32167" title="ISVNAuthentication provider did not provide credentials" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/32167">JENKINS-32167</a>"><del>JENKINS-32167</del></a></font>, there is no owner or maintainer of this plugin.  There is a workaround provided in comments under <font color="#000080"><a href="https://issue-redirect.jenkins.io/issue/32167" title="ISVNAuthentication provider did not provide credentials" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/32167">JENKINS-32167</a>"><del>JENKINS-32167</del></a></font>.  <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jglick" class="user-hover" rel="jglick">jglick</a> provided some better feedback code, but I don't think that has ever shipped.  I looked at it and wrote a test case demonstrating the problem, but solving it was too complicated for the time I could devote to it.</p>

<p>I don't think anything has changed since then.   Wish someone would step up and fix it for us users, but I'm not expecting it until the SVN plugin is totally busted.</p>

<p> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ifernandezcalvo.png" title="ifernandezcalvo's avatar" /> ifernandezcalvo:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=339669&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-339669">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jjvbintec" class="user-hover" rel="jjvbintec">jjvbintec</a> Did the latest version (2.10.6) resolve this one?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jjvbintec" class="user-hover" rel="jjvbintec">jjvbintec</a> Did the latest version (2.10.6) resolve this one?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ifernandezcalvo.png" title="ifernandezcalvo's avatar" /> ifernandezcalvo:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=341193&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-341193">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>great, I have to make a test based on those scripts.</p>
</pre>
</details>

</ul>
</details>

<p>great, I have to make a test based on those scripts.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kyates.png" title="kyates's avatar" /> kyates:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-32167?focusedId=348984&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-348984">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Similar to this issues <a href="https://issues.jenkins.io/browse/JENKINS-37642" title="SVN Checkout with external fails only after new commit" class="issue-link" data-issue-key="JENKINS-37642"><del>JENKINS-37642</del></a></p>
</pre>
</details>

</ul>
</details>

<p>Similar to this issues <a href="https://issue-redirect.jenkins.io/issue/37642" title="SVN Checkout with external fails only after new commit" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/37642">JENKINS-37642</a>"><del>JENKINS-37642</del></a></p>
